lib/serializer/BinaryDeserializer.h

```diff
@@ -43,10 +44,21 @@ class DLL_LINKAGE BinaryDeserializer : public CLoaderBase
 		Source & source;
 		std::vector<std::function<Variant()>> funcs;
 
+		template <class V>
+		struct mpl_types_impl;
+
+		template <class... Ts>
+		struct mpl_types_impl<std::variant<Ts...>> {
+			using type = boost::mpl::vector<Ts...>;
+		};
+
+		template <class V>
+		using mpl_types = typename mpl_types_impl<V>::type;
+
 		VariantLoaderHelper(Source & source):
 			source(source)
 		{
-			boost::mpl::for_each<typename Variant::types>(std::ref(*this));
+			boost::mpl::for_each<mpl_types<Variant>>(std::ref(*this));
```

Comment:
Love removing boost?
Perhaps we can replace this class with something like this:
https://stackoverflow.com/a/57643845
(and remove final usage of mpl along the way)

Reply:
About MPL - i think it is next time, because it is not a final usage.

Reply:
This will require a separate PR, not simple one as this.